python/ray/tests/test_client_reconnect.py

```diff
@@ -0,0 +1,413 @@
+from concurrent import futures
```

Comment:
Really great tests! ❤️

Reply:
I am honestly very impressed with the quality of this PR and how thoughtful @ckw017 is.